src/librustc/mir/mod.rs

```diff
@@ -1857,41 +1858,44 @@ impl<'tcx> Place<'tcx> {
     pub fn return_place() -> Place<'tcx> {
         Place {
             base: PlaceBase::Local(RETURN_PLACE),
-            projection: Box::new([]),
+            projection: List::empty(),
```

Comment:
it should be possible to use some feature gates to make `List::empty` `const fn`, thus allowing  you to resolve the FIXME above.

Reply:
Can't make `List::empty` a const fn. there's an `assert!(mem::align_of::<T>() <= 64);` so I get ...

```
error[E0019]: constant function contains unimplemented expression type
   --> src/librustc/ty/mod.rs:734:9
    |
734 |           assert!(mem::align_of::<T>() <= 64);
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ in this macro invocation
    | 
   ::: <::core::macros::builtin::assert macros>:1:1
    |
1   | / ($ cond : expr) => ({ }) ; ($ cond : expr,) => ({ }) ;
2   | | ($ cond : expr, $ ($ arg : tt) +) => ({ })
    | |__________________________________________- in this expansion of `assert!`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0019`.
error: could not compile `rustc`.
```

Reply:
And I'd also get ...

```
error[E0013]: constant functions cannot refer to statics, use a constant instead
   --> src/librustc/ty/mod.rs:735:16
    |
735 |             &*(&EMPTY_SLICE as *const _ as *const List<T>)
    |                ^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0013`.
```

Reply:
The EMPTY_SLICE could be a const fn too

Reply:
you can also just make `EMPTY_SLICE` a `const` instead of a `static`

The assert can be replaced by `[()][(mem::align_of::<T>() > 64) as usize];` (and a FIXME stating that we should use the assert again once const eval supports that)